core/controllers/suggestion.py

```diff
@@ -264,17 +264,86 @@ def post(self) -> None:
         # adding an image, there is no method to remove the uploaded image.
         # See more - https://github.com/oppia/oppia/issues/14298
         if suggestion_type != feconf.SUGGESTION_TYPE_ADD_QUESTION:
+            assert isinstance(
+                suggestion,
+                suggestion_registry.SuggestionTranslateContent
+            )
             files = self.normalized_payload.get('files')
             new_image_filenames = (
                 suggestion.get_new_image_filenames_added_in_suggestion()
             )
             if new_image_filenames and files is not None:
-                _upload_suggestion_images(
+                self._upload_new_images_in_translation(
                     files, suggestion, new_image_filenames
                 )
 
+            self._copy_images_from_target_exploration_to_translation(suggestion)
+
         self.render_json(self.values)
 
+    def _upload_new_images_in_translation(
+        self,
+        files: Dict[str, str],
+        suggestion: suggestion_registry.SuggestionTranslateContent,
+        filenames: List[str]
+    ) -> None:
+        """Saves new images introduced in translation suggestion to storage.
+
+        Args:
+            files: dict. Files containing a mapping of image
+                filename to image blob.
+            suggestion: SuggestionTranslateContent. The translation suggestion
+                for which images are being uploaded.
+            filenames: list(str). The image filenames.
+        """
+        suggestion_image_context = suggestion.image_context
+        # TODO(#10513): Find a way to save the images before the suggestion is
```

Comment:
Just to check, is this still relevant here? From reading this it's a bit unclear when the suggestion is created exactly, and what we actually want to do here.

Reply:
@seanlip Issue #10513 is still open as of now. Although I personally don't see why we should save the images before the suggestion is saved, with the exception that images for question suggestions are saved before saving the question suggestion. I think that images should be saved as the suggestion is saved, since those images are a part of the suggestion and it doesn't make sense for images uploaded by the translation submitter to exist without the suggestion existing first. What do you think?

Reply:
Yes, I think that sounds reasonable to me too. I think we can close that issue since it doesn't provide justification and we can't understand why it makes conceptual sense to do it ... I'll leave a note there.

Reply:
(Btw the comments in https://github.com/oppia/oppia/pull/10475 have some interesting context on the existing implementation, e.g. https://docs.google.com/document/d/1kGvPjYnW9ArYKtzxYOmFbp7MA-A45B3znRWFIlaEGVo/edit#heading=h.xyfn6zlrzj3e and https://github.com/oppia/oppia/pull/10475#discussion_r480376665

Reply:
@seanlip I removed the TODO comment.